File: php/features.php
<?php
declare(strict_types=1);

// ChineseHolidayCalendar SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ChineseHolidayCalendarFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ChineseHolidayCalendarBaseFeature();
            case "test":
                return new ChineseHolidayCalendarTestFeature();
            default:
                return new ChineseHolidayCalendarBaseFeature();
        }
    }
}

File: py/features.py
# ChineseHolidayCalendar SDK feature factory

from feature.base_feature import ChineseHolidayCalendarBaseFeature
from feature.test_feature import ChineseHolidayCalendarTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ChineseHolidayCalendarBaseFeature(),
        "test": lambda: ChineseHolidayCalendarTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
